File: 2-AndroidDesignPatterns/MVP/app/src/main/java/com/example/arch_lab2/MainActivity.java
package com.example.arch_lab2;


import android.content.Intent;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;

import android.view.View;
import android.widget.Button;

import com.example.arch_lab2.all_products.view.AllProductsActivity;
import com.example.arch_lab2.fav_products.view.FavProductsActivity;

public class MainActivity extends AppCompatActivity {

    Button btnAll;
    Button btnFav;
    Button btnExit;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        btnAll = findViewById(R.id.btnShowAll);
        btnFav = findViewById(R.id.btnShowFav);
        btnExit = findViewById(R.id.btnExit);
        btnAll.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, AllProductsActivity.class);
                startActivity(intent);
            }
        });
        btnFav.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, FavProductsActivity.class);
                startActivity(intent);

            }
        });
        btnExit.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
    }
}

File: 2-AndroidDesignPatterns/MVVM/app/src/main/java/com/example/mvvm/MainActivity.kt
package com.example.mvvm

import android.content.Intent
import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.example.mvvm.all_products.AllProductsActivity
import com.example.mvvm.databinding.ActivityMainBinding
import com.example.mvvm.fav_products.FavProductsActivity
import com.example.mvvm.model.local.LocalDataSource
import com.example.mvvm.model.remote.RemoteDataSource
import com.example.mvvm.model.repository.ProductRepository

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        ViewCompat.setOnApplyWindowInsetsListener(binding.main) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        binding.btnAllProducts.setOnClickListener {
            startActivity(Intent(this, AllProductsActivity::class.java))
        }

        binding.btnFavProducts.setOnClickListener {
            startActivity(Intent(this, FavProductsActivity::class.java))
        }

        binding.btnExit.setOnClickListener {
            finish()
        }

    }
}